hey-discount/src/main/java/Ads.java

```diff
@@ -0,0 +1,121 @@
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+public class Ads {
+    private List<Ad> ads;
+
+    public Ads(){
+        ads = new ArrayList<>();
+    }
+
+    public Ad getAd(String id){
+        for (Ad ad : ads) {
+            if(ad.getId().equals(id)) {
+                return ad;
+            }
+        }
+        return null;
+    }
+
+    public boolean validate (Ad ad){
+        if (ad.getId() == null || ad.getId().length() < 1) {
+            return false;
+        }
+        if (ad.getDescription() == null || ad.getDescription().length() > 200) {
+            return false;
+        }
+        if (ad.getCreatedAt() == null) {
+            return false;
+        }
+        if (ad.getVendor() == null || ad.getVendor().length() < 1) {
+            return false;
+        }
+        if (ad.getLink() == null || ad.getLink().length() < 1) {
+            return false;
+        }
+        if (ad.getHashTags().isEmpty()) {
+            return false;
+        }
+        if (ad.getDiscount() == null || ad.getDiscount().length() < 1) {
+            return false;
+        }
+        if (ad.getValidUntil() == null) {
+            return false;
+        }
+        return true;
+    }
+
+    public boolean addAd(Ad adToAdd){
+        for (Ad ad : ads) {
+            if(ad.getId().equals(adToAdd.getId())) {
+                return false;
+            }
+        }
+        if (!validate(adToAdd)) {
+            ads.add(adToAdd);
+            return false;
+        }
+        return true;
+    }
+
+    public boolean editAd(String id, Ad ad){
+        Ad adForEdit = this.getAd(id);
```

Comment:
Maybe just remove an ad by `id` from the collection and add the passed one to the collection?